### ISSUE TITLE
[HDTorrents.it] No results found

### ISSUE THREAD

**Issue by EffeF** [Jackett/Jackett#1294]
I've inserted between the indexers of Sonarr and Radarr, the test is ok, but then it does not find any result.

Jackett version:  
v0.7.1296

**Comment by Oscarteg** [comment]
Do you by any chance use Docker for these applications?

**Comment by EffeF** [comment]
No, are installed on Windows 10

**Comment by kaso17** [comment]
please provide an advacned log

**Comment by EffeF** [comment]
Where do I find the log file?

**Comment by kaso17** [comment]
Try C:\ProgramData\Jackett\log.txt on windows

**Comment by EffeF** [comment]
This?
[log.txt](https://github.com/Jackett/Jackett/files/943165/log.txt)

**Comment by kaso17** [comment]
It's some lower level owin exception, probably not much we can do about. With an english error message we might have more luck. 
Are you using any security/antivirus/firewall products?

**Comment by EffeF** [comment]
Avira free... but can it cause problems with just one tracker? With all the others I have no problems.

**Comment by kaso17** [comment]
In the logs I can see the errors just for limetorrents and skytorrents. It's a socket related error, various security software is messing with the windows sockets to scan the network traffic, etc. 
I suggest you try to turn of avira and see if it changes anything.

**Comment by EffeF** [comment]
But limetorrents and skytorrents do not give me any trouble... I've tried a tracker at a time and this is the result (manual search on Jackett and on Radar).

LimeTorrents:
<img src="https://anonimag.es/i/1a2f4d5.png" alt="1a2f4d5.png" border="0">
<img src="https://anonimag.es/i/1b9330d.png" alt="1b9330d.png" border="0">

Sky torrents:
<img src="https://anonimag.es/i/2a0893b.png" alt="2a0893b.png" border="0">
<img src="https://anonimag.es/i/2b83bdd.png" alt="2b83bdd.png" border="0">

HDTorrents:
<img src="https://anonimag.es/i/3a336ec.png" alt="3a336ec.png" border="0">
<img src="https://anonimag.es/i/3b2c6f7.png" alt="3b2c6f7.png" border="0">

I first turned off and then uninstalled Avira, but the result does not change

**Comment by kaso17** [comment]
Please provide the jackett log showing these particular hdtorrents.it search results (with enhanced logging enabled).

**Comment by EffeF** [comment]
[log.txt](https://github.com/Jackett/Jackett/files/946709/log.txt)

**Comment by kaso17** [comment]
In your manual search query you're just searching for "Collateral Beauty". Radarr searches for "Collateral Beauty 2016". That doesn't seem to work. Can you search for "Collateral Beauty 2016" directly on the trackers?

**Comment by EffeF** [comment]
No result:
<img src="https://anonimag.es/i/4de0f9.png" alt="4de0f9.png" border="0">


but adding the year I don't have any results not even on the tracker website:
<img src="https://anonimag.es/i/4bcbfd8.png" alt="4bcbfd8.png" border="0"><img src="https://anonimag.es/i/4c3271d.png" alt="4c3271d.png" border="0">

**Comment by EffeF** [comment]
@bonny1992 can you help me?

**Comment by bubez81** [comment]
Same problem here no results from hdtorrent.it

**Comment by zodiacuhd** [comment]
Same here, HDtorrent.it does not seem to work.

**Comment by bubez81** [comment]
now it give me errors
[hdtorrentlog.pdf](https://github.com/Jackett/Jackett/files/1242108/hdtorrentlog.pdf)

**Comment by iFelix18** [comment]
Same problem here too, I can not use HDTorrents on Radarr.

**Comment by EffeF** [comment]
Please, help me.
HDTorrents is the best Italian tracker for movies, without it I can not use Jackett and Radarr.

I reported the issue five months ago, and since then nothing has changed.

**Comment by zodiacuhd** [comment]
I do agree, HDTorrents.it is the only proper italian tracker and the fact that it isn't working, is a pity.

Here is one of the latest errors showing in the logs. If we can do anything for the developers, please let us know.
[HDTorrents.it log.txt](https://github.com/Jackett/Jackett/files/1309253/HDTorrents.it.log.txt)

**Comment by EffeF** [comment]
I found the problem for the movies, maybe...

Take for example the movie _**"La Mummia (2017)"**_, Radarr search:
`la mummia 2017` title in italian with year
`mummy 2017`  title in English with year, but without parenthesis
or
`mummy 3d 2017`
but the tracker, on the site, only accepts: 
`La mummia` title in Italian without a year
or
`The Mummy (2017)` title in English with year in parenthesis

Instead we take the film _**"Pirati Dei Caraibi - La Vendetta Di Salazar (2017)"**_, Radarr search:
`pirati dei caraibi la vendetta di salazar 2017` title in italian with year, but without the dash
`pirates caribbean dead men tell no tales 2017` title in English with year, but without parenthesis and the two points
`pirates caribbean 5 2017`
`pirates caribbean 5 dead men tell no tales 2017`
`pirates caribbean 5 dead men tell no tales 2017 2017`
but the tracker, on the site, only accepts: 
`Pirati dei Caraibi - La vendetta di Salazar` title in Italian without a year, and with the dash
or
`Pirates of the Caribbean: Dead Men Tell No Tales (2017)` title in English with year in parenthesis, and with the two points

Taking the title in English, and putting in parentheses the year would find the majority of the films. The problem remains with the titles with "-" and ":"

I hope this information can help solve the problem.

**Comment by EffeF** [comment]
I did as written here: #1636 

The modified code is:

  search:
    paths:
      - path: browse.php
        method: post
    keywordsfilters:
      - name: re_replace
        args: ["([1-2]\\d{3})", ""]
    inputs:
      ajax: false
      search: "{{ .Keywords }}"
      incldead: 0

I also had to enable the option "Lenient mapping" in Radarr:
![2](https://user-images.githubusercontent.com/25557983/30775460-4ac71060-a094-11e7-84f3-f68544b7f0a8.PNG)

It does not work with movies that have the dash or the two dots, such as with **_"Pirati Dei Caraibi - La Vendetta Di Salazar (2017)"_**.

With the film **_"La Mummia (2017)"_** this is the results:
![1 0](https://user-images.githubusercontent.com/25557983/30775506-41429216-a095-11e7-9272-235753dd9cb2.PNG)
![1 1](https://user-images.githubusercontent.com/25557983/30775508-46e98346-a095-11e7-9660-907a13a66553.PNG)
He does not find the torrent quality, nor the Italian language, but at least he finds the movie

**Comment by EffeF** [comment]
I made another change:
```
    fields:
      title-no-filters:
        selector: a[href^="download.php?id="]
        attribute: href
      title:
        selector: a[href^="download.php?id="]
        attribute: href
        filters:
          - name: querystring
            args: name
          - name: re_replace
            args: [".torrent", ""]
```
Now with the function "Strict" active recognizes the title, the Italian language, and the quality of the film, unfortunately just still can not find the movies with a titles that contain `- ` or `:`.

I also have to test if it works with Sonarr and the TV series.

**Comment by EffeF** [comment]
I made some changes, can anyone test how it works?
[hdtorrentsit.zip](https://github.com/Jackett/Jackett/files/1329454/hdtorrentsit.zip)

Known Issues:
- it does not work with TV series and Sonar
- does not find the movies that contain `-` or `:` in the title
-  it does not work with movies that do not have audio tags in the name of the torrent file

from some tests I've done now Radarr finds most of the films

**Comment by rafuz** [comment]
I will test it and report what I find.
Thank you!

**Comment by rafuz** [comment]
Still issues with some movie titles and year

![image](https://user-images.githubusercontent.com/22271829/31012127-f925b712-a510-11e7-93d0-d7bb73a0e1f4.png)

Removing year it works;

![image](https://user-images.githubusercontent.com/22271829/31012150-0e9f5940-a511-11e7-979f-9f0f02f3bad3.png)

The problem is Radarr always searches with years in the keywords.

**Comment by EffeF** [comment]
By searching directly on the HDTorrent site `Italian Title + Year` no results are obtained, only accepts `English Title + Year`. I might try to remove the year, but you would get wrong results if more movies contain the same keyword. So I made sure that Jackett worked with the original title.

The problem is that the tracker only accepts the exact title:
"Fantastic Beasts **And** Where To Find Them"
but Radarr search:
"Fantastic Beasts Where To Find Them" (without **And**)

It is the usual problem of the films that contain `-` and `:` in the title, Radarr removes them from search.
In summary, it does not work with titles that contain `-`, `:` and `and`... but I think this is a Radarr problem.

I will try to find a solution

**Comment by EffeF** [comment]
Try this: [hdtorrentsit.zip](https://github.com/Jackett/Jackett/files/1345010/hdtorrentsit.zip)

Now it should work, I've eliminated the year from research.
The problem is that you can now find wrong results (try with `La Mummia 2017` for example), but then the results will be filtered by Radarr, which will download the right release

**Comment by rafuz** [comment]
Ok, I will do some checks but I think should work.
Anyway sonarr removes articles and conjuction when searching so ther are a lot of movies that will never match on hdtorrent.

**Comment by EffeF** [comment]
I made some other changes: [hdtorrentsit.zip](https://github.com/Jackett/Jackett/files/1346508/hdtorrentsit.zip)


Now it works with movies that with most of the series.
There remains the problem of the titles with "-" and ":",  and has problem with some old tv series that did not follow the most popular naming scheme on HDTorrents.

If it's all ok I proceed with a pull request.

**Comment by kaso17** [comment]
Regarding the - or : issue:
I suggest you try the tracker search with wildcards.
e.g. 
*Pirati*Dei*Caraibi*La*Vendetta*Di*Salazar*
%Pirati%Dei%Caraibi%La%Vendetta%Di%Salazar%

**Comment by EffeF** [comment]
Yes, it works with:
`%Pirati%Dei%Caraibi%La%Vendetta%Di%Salazar%`
thanks for the tip.

Updated version: [hdtorrentsit.zip](https://github.com/Jackett/Jackett/files/1347270/hdtorrentsit.zip)

Let me know if now everything is ok.

**Comment by rafuz** [comment]
It seems really nice now:

![image](https://user-images.githubusercontent.com/22271829/31057715-b7e4d5ca-a6e7-11e7-8d0f-1ed5096704d5.png)
